rosco-core/src/main/groovy/com/netflix/spinnaker/rosco/providers/alicloud/AliCloudBakeHandler.java

```diff
@@ -0,0 +1,211 @@
+/*
+ * Copyright 2019 Alibaba Group, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.rosco.providers.alicloud;
+
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializationFeature;
+import com.netflix.spinnaker.rosco.api.Bake;
+import com.netflix.spinnaker.rosco.api.BakeOptions;
+import com.netflix.spinnaker.rosco.api.BakeOptions.BaseImage;
+import com.netflix.spinnaker.rosco.api.BakeRequest;
+import com.netflix.spinnaker.rosco.providers.CloudProviderBakeHandler;
+import com.netflix.spinnaker.rosco.providers.alicloud.config.RoscoAliCloudConfiguration;
+import com.netflix.spinnaker.rosco.providers.alicloud.config.RoscoAliCloudConfiguration.AliCloudOperatingSystemVirtualizationSettings;
+import com.netflix.spinnaker.rosco.providers.alicloud.config.RoscoAliCloudConfiguration.AliCloudVirtualizationSettings;
+import com.netflix.spinnaker.rosco.providers.util.ImageNameFactory;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Component
+public class AliCloudBakeHandler extends CloudProviderBakeHandler {
+
+  private static final String IMAGE_NAME_TOKEN = "alicloud-ecs: Creating image:";
+
+  ImageNameFactory imageNameFactory = new ImageNameFactory();
+
+  @Autowired RoscoAliCloudConfiguration.AliCloudBakeryDefaults alicloudBakeryDefaults;
```

Comment:
I think if it use constructor injection, there will be "Loop nesting" issue. Do you have some ideas?

Reply:
I haven't run into that before, maybe there's a circular dependency here?  Do you have details on the loop nesting error you got when you tried this?

Reply:
HI @ezimanyi This is the error message:
```
Description:

The dependencies of some of the beans in the application context form a cycle:

┌─────┐
|  aliCloudBakeHandler defined in file [/Users/luoguan01/Desktop/test/rosco/rosco-core/out/production/classes/com/netflix/spinnaker/rosco/providers/alicloud/AliCloudBakeHandler.class]
↑     ↓
|  com.netflix.spinnaker.rosco.providers.alicloud.config.RoscoAliCloudConfiguration
└─────┘
```

Reply:
OK, I understand now---there is indeed a circular dependency.  It seems strange that `RoscoAliCloudConfiguration` should depend on `AliCloudBakeHandler`.  It looks like this dependency is only so that we can call `cloudProviderBakeHandlerRegistry.register` in a `@PostConstruct`.

Ideally we'd remove this circular dependency completely (maybe by having the bake handler directly register itself rather than via the configuration class).  I'll let you decide if you want to make that change, or leave it as is.  (It looks like this issue is present in all the bake handlers.)

Reply:
Thanks  a lot